inc/umock_c/umock_c_internal.h

```diff
@@ -957,134 +994,231 @@ typedef struct MOCK_CALL_METADATA_TAG
+        MU_FOR_EACH_2_KEEP_1(DECLARE_IGNORE_ARGUMENT_FUNCTION_PROTOTYPE, name, __VA_ARGS__) \
+        MU_FOR_EACH_2_KEEP_1(DECLARE_VALIDATE_ARGUMENT_FUNCTION_PROTOTYPE, name, __VA_ARGS__) \
+        MU_FOR_EACH_2_KEEP_1(DECLARE_COPY_OUT_ARGUMENT_BUFFER_FUNCTION_PROTOTYPE, name, __VA_ARGS__) \
+        MU_FOR_EACH_2_KEEP_1(DECLARE_VALIDATE_ARGUMENT_VALUE_FUNCTION_PROTOTYPE, name, __VA_ARGS__) \
+        MU_FOR_EACH_2_KEEP_1(DECLARE_VALIDATE_ARGUMENT_VALUE_AS_TYPE_FUNCTION_PROTOTYPE, name, __VA_ARGS__) \
+        MU_FOR_EACH_2_KEEP_1(DECLARE_CAPTURE_ARGUMENT_VALUE_FUNCTION_PROTOTYPE, name, __VA_ARGS__) \
+        typedef struct MU_C2(_mock_call_modifier_,name) (*MU_C2(ignore_one_argument_func_type_,name))(void); \
+    ,) \
+    MU_IF(MU_COUNT_ARG(__VA_ARGS__), \
+        extern const MU_C2(ignore_one_argument_func_type_,name) MU_C2(ignore_one_argument_array_,name)[];\
+    ,) \
     DECLARE_VALIDATE_ONE_ARGUMENT_FUNC_TYPE(name) \
-    static const MU_C2(validate_one_argument_func_type_,name) MU_C2(validate_one_argument_array_,name)[] = \
-    {,) \
+    extern const MU_C2(validate_one_argument_func_type_,name) MU_C2(validate_one_argument_array_,name)[]; \
+    void MU_C2(fill_mock_call_modifier_, name)(MU_C2(mock_call_modifier_, name)* mock_call_modifier);\
+    typedef struct MU_C2(_mock_call_, name) \
+    { \
+        MU_IF(IS_NOT_VOID(return_type), \
+            return_type return_value; \
+            return_type fail_return_value; \
+            return_type* captured_return_value; \
+        ,) \
+        MU_IF(MU_COUNT_ARG(__VA_ARGS__), \
+            MU_FOR_EACH_2(DECLARE_MOCK_CALL_STRUCT_STACK, __VA_ARGS__) \
+            MU_FOR_EACH_2(DECLARE_OUT_ARG_BUFFER_FOR_ARG, __VA_ARGS__) \
+            ARG_BUFFER* out_arg_buffers[MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_DIV2(MU_COUNT_ARG(__VA_ARGS__)), 1)]; \
+            ARG_BUFFER validate_arg_buffers[MU_IF(MU_COUNT_ARG(__VA_ARGS__), MU_DIV2(MU_COUNT_ARG(__VA_ARGS__)), 1)]; \
+            MU_FOR_EACH_2(DECLARE_VALIDATE_ARG_VALUE, __VA_ARGS__) \
+            MU_FOR_EACH_2(DECLARE_CAPTURE_ARG_VALUE, __VA_ARGS__) \
+            MU_FOR_EACH_2(DECLARE_IGNORE_FLAG_FOR_ARG, __VA_ARGS__) \
+            MU_FOR_EACH_2(DECLARE_OVERRIDE_ARGUMENT_TYPE_FOR_ARG, __VA_ARGS__)  \
+        ,) \
+        MU_IF(IS_NOT_VOID(return_type), \
+            int fail_return_value_set : 1; \
+            int return_value_set : 1;   \
+        ,) \
+        MU_IF(MU_COUNT_ARG(__VA_ARGS__), \
+            , \
+            MU_IF(IS_NOT_VOID(return_type), \
+                , \
+                int dummy : 1;\
+            )\
+        ) \
+    } MU_C2(mock_call_, name); \
+    typedef MU_C2(mock_call_modifier_,name) (*MU_C3(auto_ignore_args_function_,name,_type))(MU_C2(mock_call_modifier_,name) call_modifier, const char* call_as_string); \
+    MU_C2(mock_call_modifier_,name) MU_C2(auto_ignore_args_function_,name)(MU_C2(mock_call_modifier_,name) call_modifier, const char* call_as_string); \
```

Comment:
These macros are almost like ... "if it compiles it runs" :-)

Reply:
"we have tests" :)